ci/JENKINS.md

```diff
@@ -0,0 +1,88 @@
+
+## Service account
+
+The Jenkins server must be configured with a GCP service account with the following roles:
+
+- Cloud Functions Developer - roles/cloudfunctions.developer
+- Compute Network Viewer - roles/compute.networkViewer
+- Kubernetes Engine Admin - roles/container.admin 
+- Create Service Accounts - roles/iam.serviceAccountCreator
+- Pub/Sub Editor - roles/pubsub.editor
+- Cloud Memorystore Redis Admin - roles/redis.admin
+- Service Usage Admin - roles/serviceusage.serviceUsageAdmin
+- Source Repository Administrator - roles/source.admin
+- Storage Admin - roles/storage.admin
+- Project IAM Admin - roles/resourcemanager.projectIamAdmin
+- Service Account User - roles/iam.serviceAccountUser
+
+These roles can be enabled through the IAM UI or with [gcloud](https://cloud.google.com/sdk/gcloud/reference/projects/add-iam-policy-binding).
+
+## Properties file
+
+In order to install Spinnaker for GCP, you must provide the installation script with a properties file. This properties file can be generated by executing [setup_properties.sh](../scripts/install/setup_properties.sh). Once you have a properties file, it must be provided to the installation script during the execution of the Jenkins job.
+
+### Jenkins Job Configuration
+
+Once the aforementioned dependencies have been fulfilled, follow the below steps to configure a job for installing Spinnaker for GCP. 
+
+1. Create a `New Item` from the Jenkins menu and select a `Freestyle project`. 
+2. From the configuration screen, enable `Color ANSI Console Output` and set the `ANSI color map` to `xterm`.
+3. Add an `Execute Shell` build step.
+4. Configure the build step to retrieve and execute the `setup.sh` script.
+
+```shell
+#!/usr/bin/env bash
+
+set -e
+
+git clone https://github.com/GoogleCloudPlatform/spinnaker-for-gcp.git
+git config --global user.name "jenkins-user"
+git config --global user.email "jenkins-user@example.com"
+
+REPO_PATH=$WORKSPACE PROPERTIES_FILE=$PROPERTIES CI=true $WORKSPACE/spinnaker-for-gcp/scripts/install/setup.sh
+```
+
+In the above example, the git `user.name` and `user.email` are configured prior to running `setup.sh`. This step is required for `setup.sh` to run successfully. Git operations can also be managed using the [Jenkins Git plugin](https://plugins.jenkins.io/git).
+
+`setup.sh` requires several variables to be passed in:
+
+- `REPO_PATH`: The absolute path for the Jenkins workspace. Jenkins makes this available via `$WORKSPACE`.
+- `PROPERTIES_FILE`: This is the absolute path to the your generated Spinnaker for GCP properties file.
+- `CI`: This must be set to `true` when running `setup.sh` outside of Cloud Shell.
+
+Once configured, execute the job to install Spinnaker for GCP. Changes to the properties file can be applied by re-running the job. Additional instructions for how to access or manage the deployed Spinnaker application are available [here](https://cloud.google.com/docs/ci-cd/spinnaker/spinnaker-for-gcp#access_spinnaker).
```

Comment:
```suggestion
1. Execute the job to install Spinnaker for GCP. 

If you change the properties file, apply the change by re-running the job. 

Additional instructions for how to access or manage the deployed Spinnaker application are available [here](https://cloud.google.com/docs/ci-cd/spinnaker/spinnaker-for-gcp#access_spinnaker).
```

Reply:
If you indent everything between step 4 and this step (Execute the job...), by three spaces, this becomes just another step in the procedure.